File: apps/high-steak-api/src/main/java/com/highsteak/api/validation/ApiConstraints.java
package com.highsteak.api.validation;

/**
 * HTTP validation limits aligned with database column definitions in Flyway migrations.
 */
public final class ApiConstraints {

    private ApiConstraints() {}

    public static final int USERNAME_MIN = 3;
    public static final int USERNAME_MAX = 50;

    public static final int EMAIL_MAX = 255;

    public static final int PASSWORD_MIN = 8;
    public static final int PASSWORD_MAX = 100;

    public static final int DISPLAY_NAME_MIN = 2;
    public static final int DISPLAY_NAME_MAX = 100;

    public static final int POST_TITLE_MIN = 1;
    public static final int POST_TITLE_MAX = 120;

    /** MySQL {@code TEXT} column limit for {@code steak_posts.comment}. */
    public static final int POST_COMMENT_MAX = 65_535;

    public static final int RESTAURANT_NAME_MAX = 120;
    public static final int RESTAURANT_LOCATION_MAX = 255;

    /** MySQL {@code TEXT} column limit for {@code post_comments.body}. */
    public static final int COMMENT_BODY_MAX = 2_000;

    public static final int SEARCH_QUERY_MIN = 2;
    public static final int SEARCH_QUERY_MAX = 100;

    public static final int MAX_REVIEW_TAGS = 12;

    public static final long MAX_IMAGE_BYTES = 1_048_576L;
}

File: apps/high-steak-mobile/lib/constants/api_constraints.dart
/// Field limits aligned with the API OpenAPI spec and database columns.
/// Image size default: 5 MB. Override at build time: `--dart-define=MAX_IMAGE_SIZE_MB=5`
/// At runtime the app also loads the live limit from `GET /api/config`.
class ApiConstraints {
  ApiConstraints._();

  static int _maxImageMb = int.fromEnvironment('MAX_IMAGE_SIZE_MB', defaultValue: 5);

  static int get maxImageMb => _maxImageMb;

  static void applyRemoteConfig(int maxImageSizeMb) {
    if (maxImageSizeMb > 0) {
      _maxImageMb = maxImageSizeMb;
    }
  }

  static int get maxImageBytes => maxImageMb * 1048576;

  static const usernameMin = 3;
  static const usernameMax = 50;
  static const emailMax = 255;
  static const passwordMin = 8;
  static const passwordMax = 100;
  static const displayNameMin = 2;
  static const displayNameMax = 100;
  static const postTitleMax = 120;
  static const postCommentMax = 65535;
  static const restaurantNameMax = 120;
  static const restaurantLocationMax = 255;
  static const commentBodyMax = 2000;
  static const searchQueryMin = 2;
  static const searchQueryMax = 100;
  static const maxReviewTags = 12;
  static const maxImagesPerPost = 10;
}
